File: frontend/src/types/product.ts
export interface SitePrice {
  site: string;
  site_display_name: string;
  url: string;
  price: number;
  currency: string;
  in_stock: boolean;
  last_updated: string | null;
}

export interface Product {
  id: string;
  name: string;
  normalized_name: string;
  brand: string;
  category: string;
  specs: Record<string, string>;
  site_prices: SitePrice[];
  min_price: number | null;
  min_price_site: string | null;
  image_url: string | null;
  last_scraped: string | null;
  created_at: string | null;
  updated_at: string | null;
}

export interface ProductListResponse {
  items: Product[];
  total: number;
  page: number;
  pages: number;
}

export interface ProductFilters {
  category?: string;
  site?: string;
  brand?: string;
  min_price?: number;
  max_price?: number;
  in_stock?: boolean;
  sort?: string;
  page?: number;
  per_page?: number;
}

File: backend/api/models/product.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Any


class SitePrice(BaseModel):
    site: str
    site_display_name: str
    url: str
    price: float
    currency: str = "TRY"
    in_stock: bool
    last_updated: datetime | None = None


class ProductResponse(BaseModel):
    id: str
    name: str
    normalized_name: str
    brand: str = ""
    category: str
    specs: dict[str, Any] = {}
    site_prices: list[SitePrice] = []
    min_price: float | None = None
    min_price_site: str | None = None
    image_url: str | None = None
    last_scraped: datetime | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None


class ProductListResponse(BaseModel):
    items: list[dict]
    total: int
    page: int
    pages: int
